staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -161,13 +162,13 @@ type SharedInformer interface {
 	GetStore() Store
 	// GetController is deprecated, it does nothing useful
 	GetController() Controller
-	// Run supports the legacy way interface for running shared index informers by
-	// merging the provided stopCh with the internal stopRead on the informer
-	// and calling RunWithStopOptions.
-	Run(stopCh <-chan struct{})
 	// RunWithStopOptions starts and runs the shared informer, returning after it stops.
 	// The informer will be stopped when a stopOptions condition is met.
 	RunWithStopOptions(ctx context.Context, stopOptions StopOptions)
+	// Run supports the legacy way interface for running shared index informers
+	// with just a stop channel.
+	// Deprecated: use RunWithStopOptions instead.
```

Comment:
same question as that for reflector:
have we agreed to deprecate it? it seems to have its own use case, and good to keep as a convenience wrapper around the RunWithStopOptions.

Reply:
Correct, we have not agreed to deprecate it, I have removed the comment. Initially I added it because there is no functional difference between calling Run and RunWithStopOptions (with stopOptions that do not shut the informer down), but the former is being kept in order to not break existing clients. I think it is useful to recommend new clients to use RunWithStopOptions, but you are correct that we have not agreed to deprecate it.